src/Serializing/SExpression/Precise.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DataKinds, FlexibleContexts, FlexibleInstances, MultiParamTypeClasses, OverloadedStrings, ScopedTypeVariables, TypeApplications, TypeFamilies, TypeOperators, UndecidableInstances #-}
+module Serializing.SExpression.Precise
+( serializeSExpression
+) where
+
+import Data.ByteString.Builder
+import Data.Foldable (fold)
+import Data.List (intersperse)
+import Data.Text (Text)
+import GHC.Generics
+
+serializeSExpression :: ToSExpression t => t -> Builder
+serializeSExpression t = toSExpression t 0 <> "\n"
```

Comment:
I’ve used the same name for the class as in `Serializing.SExpression` because I intend that this module should eventually replace the former (and in the meantime, module namespacing is well and good).

This implementation does _not_ support `Options` to show only the annotations of nodes, which we use in `Serializing.SExpression` for `ts-parse`, which I think we should drop as part of the migration to precise ASTs.